map/bookmark_manager.cpp

```diff
@@ -1871,7 +1871,17 @@ void BookmarkManager::LoadBookmarkRoutine(std::string const & filePath, bool isT
     std::string fileSavePath = GetKMLPath(filePath);
     if (!fileSavePath.empty())
     {
-      auto kmlData = LoadKmlFile(fileSavePath, KmlFileType::Text);
+      auto const ext = base::GetFileExtension(filePath);
+      std::unique_ptr<kml::FileData> kmlData;
+      if (ext == ".gpx")
+      {
+        kmlData = LoadKmlFile(fileSavePath, KmlFileType::Gpx);
```

Comment:
What are the benefits of storing files internally in different formats? Why is it a bad idea to convert imported GPX files to KML ones and store them internally in KML?

Reply:
I don't think that different formats are used internally, this parameter is used just to select a valid parser:

i```
f (fileType == KmlFileType::Binary)
    {
      kml::binary::DeserializerKml des(*data);
      des.Deserialize(reader);
    }
    else if (fileType == KmlFileType::Text)
    {
      kml::DeserializerKml des(*data);
      des.Deserialize(reader);
    }
    else if (fileType == KmlFileType::Gpx)
    {
      kml::DeserializerGpx des(*data);
      des.Deserialize(reader);
    }
```